driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTest.java

```diff
@@ -84,15 +95,35 @@ public abstract class UnifiedTest {
     private final BsonArray entitiesArray;
     private final BsonArray initialData;
     private final BsonDocument definition;
-    private final AssertionContext context = new AssertionContext();
     private final Entities entities = new Entities();
     private final UnifiedCrudHelper crudHelper;
     private final UnifiedGridFSHelper gridFSHelper = new UnifiedGridFSHelper(entities);
     private final UnifiedClientEncryptionHelper clientEncryptionHelper = new UnifiedClientEncryptionHelper(entities);
-    private final ValueMatcher valueMatcher = new ValueMatcher(entities, context);
-    private final ErrorMatcher errorMatcher = new ErrorMatcher(context);
-    private final EventMatcher eventMatcher = new EventMatcher(valueMatcher, context);
     private final List<FailPoint> failPoints = new ArrayList<>();
+    private final UnifiedTestContext rootContext = new UnifiedTestContext();
+
+    private class UnifiedTestContext {
```

Comment:
This refactoring was required in order to support runOnThread.  The problem is that the context is not really capable of supporting concurrent operations, so what we do now is create a new instance of UnifiedTestContext in runOnThread, and  then pass UnifiedTestContext around everywhere instead of relying on the root context.